src/main/kotlin/no/nav/familie/ef/sak/iverksett/oppgaveforbarn/BarnFyllerÅrOppfølgingsoppgaveService.kt

```diff
@@ -0,0 +1,142 @@
+        }
+    }
+
+    private fun filtrerBarnSomHarFyltÅr(barnTilUtplukkForOppgave: List<BarnTilUtplukkForOppgave>): List<OpprettOppgaveForBarn> {
+        val opprettedeOppgaver = oppgaveRepository.findByTypeAndAlderIsNotNull(Oppgavetype.InnhentDokumentasjon)
+        val skalOpprettes = mutableListOf<OpprettOppgaveForBarn>()
+
+        barnTilUtplukkForOppgave.forEach { barn ->
+            val barnetsAlder = Alder.fromFødselsdato(fødselsdato(barn))
+            if (barnetsAlder != null && barn.fødselsnummerBarn != null && opprettedeOppgaver.none { it.barnPersonIdent == barn.fødselsnummerBarn && it.alder == barnetsAlder }) {
+                skalOpprettes.add(
+                    OpprettOppgaveForBarn(
+                        barn.fødselsnummerBarn,
+                        barn.fødselsnummerSøker,
+                        barnetsAlder,
+                        barn.behandlingId
+                    )
+                )
+            }
+        }
+
+        logger.info("barn til utplukk for oppgave: ${skalOpprettes.size}")
+        return skalOpprettes
+    }
+
+    private fun opprettOppgaveForBarn(opprettOppgaverForBarn: List<OpprettOppgaveForBarn>) {
+        if (opprettOppgaverForBarn.isEmpty()) return
+        val gjeldendeBarnList =
+            gjeldendeBarnRepository.finnEksternFagsakIdForBehandlingId(opprettOppgaverForBarn.map { it.behandlingId }).toSet()
+        gjeldendeBarnList.forEach { gjeldendeBarn ->
+            val opprettOppgaveForEksternId =
+                opprettOppgaverForBarn.firstOrNull { it.fødselsnummer == gjeldendeBarn.barnPersonIdent }
+            val finnesOppgave = oppgaveRepository.findByBehandlingIdAndBarnPersonIdentAndAlder(
+                gjeldendeBarn.behandlingId,
+                gjeldendeBarn.barnPersonIdent,
+                opprettOppgaveForEksternId?.alder
+            ) != null
+            if (!finnesOppgave && opprettOppgaveForEksternId != null) {
+                val opprettOppgaveRequest = lagOppgaveRequestForOppfølgingAvBarnFyltÅr(opprettOppgaveForEksternId, gjeldendeBarn)
+                val opprettetOppgaveId = oppgaveClient.opprettOppgave(opprettOppgaveRequest)
+                oppgaveClient.leggOppgaveIMappe(opprettetOppgaveId)
+                val oppgave = Oppgave(
+                    gsakOppgaveId = opprettetOppgaveId,
+                    behandlingId = gjeldendeBarn.behandlingId,
+                    barnPersonIdent = gjeldendeBarn.barnPersonIdent,
+                    type = Oppgavetype.InnhentDokumentasjon,
+                    alder = opprettOppgaveForEksternId.alder
+                )
+                oppgaveRepository.insert(oppgave)
+            }
+        }
+    }
+
+    private fun lagOppgaveRequestForOppfølgingAvBarnFyltÅr(
+        opprettOppgaveForEksternId: OpprettOppgaveForBarn,
+        barnTilOppgave: BarnTilOppgave
+    ) =
+        OpprettOppgaveRequest(
+            ident = OppgaveIdentV2(
+                ident = opprettOppgaveForEksternId.fødselsnummer,
```

Comment:
Her opprettes det oppgave på barnets fødselsnummer, burde det være søkerens? 
Kan vi bruke `opprettOppgave` i `OppgaveService` for dette?

Reply:
Er nok søkerens fnr som må brukes til oppgave, ja.

Reply:
Rettet